### PR TITLE
VCRUISE: Only require MP3 and JPEG support for game versions that need it

### PR DESCRIPTION
**Pull request by ccawley2011**

### CODE REVIEW

File: engines/vcruise/detection.h [hunk 1]
```diff
@@ -26,17 +26,25 @@
...
+
+	REQUIRES_MP3 = (1 << 0),
+	REQUIRES_JPEG = (1 << 1)
+};
+
 struct VCruiseGameDescription {
 	ADGameDescription desc;
 
 	VCruiseGameID gameID;
+	byte requirements;
```
Comment (sev-) [inline engines/vcruise/detection.h:47]: Why don't you make game flags? E.g. stuff it in `desc.flags` ?

**Comment by elasota** [comment]
I made a mistake with the initial report of this problem anyway: Schizm does not use MP3, it uses Vorbis.

File: engines/vcruise/vcruise.cpp [hunk 2]
```diff
@@ -65,6 +66,18 @@ void VCruiseEngine::handleEvents() {
 }
 
 Common::Error VCruiseEngine::run() {
+#ifndef USE_JPEG
+	if (_gameDescription->gameID & REQUIRES_JPEG) {
```
Comment (elasota) [inline engines/vcruise/vcruise.cpp:70]: These are checking the gameID field instead of requirements

**Review by elasota — CHANGES_REQUESTED** [review]

**Comment by elasota** [comment]
Since pretty much everything in this needed some revision I just did it myself, equivalent changes are in as of f21ea7ba3e55f62c870600c21d5f918c423051f4

Closing this.

**Pull request closed without merging**